recommendations.md

```diff
@@ -0,0 +1,236 @@
+more time.
+
+### Bad:
+
+| K8s version | AKS GA   | End of life |
+|-------------|----------|-------------|
+| 1.22        | Nov 2021 | 1.25 GA     |
+| 1.23        | Feb 2022 | 1.26 GA     |
+
+(Source: [Azure Kubernetes Release Calendar][aks])
+
+### Bad:
+
+|Version|Release Date|
+---|---
+2.1|3rd March 2021
+2.0|1st March 2020
+
+Release are supported for 2 years from the release date.
+
+### Good
+
+Same as above, but we do the math:
+
+Version|Release Date|EoL Date
+---|---|---
+2.1|3rd March 2021|3rd March 2023
+2.0|1st March 2020|1st March 2022
+
+
+### Good:
+
+|Kubernetes version|Upstream release|Amazon EKS release|Amazon EKS end of support|
+|------|-------------------|-------------------|--------------------|
+| 1.20 | December 8, 2020  | May 18, 2021      | July, 2022         |
+| 1.21 | April 8, 2021     | July 19, 2021     | September, 2022    |
+
+(Source: [Amazon EKS Release Calendar][eks])
+
+## Provide Complete Date
+
+Always document complete dates, instead of just providing a month and year.
+Users should not be left guessing whether the EoL is on the 1st of December or 31st.
+
+Bad: See above AKS and EKS examples.
+
+## Provide a release schedule image
+
+This is optional, but a clear graphical representation of release cycles (with different colors for different levels of support) is always
+nice to have.
+
+- Label your axes clearly with year boundaries.
+- Have a straight line marking the current date.
+- If you can make it interactive, provide a start and end date on hover.
+- Make sure these images are kept updated - it is easy for them to get out of date.
+- Ensure that all the data in the image is also reflected in text (in a table) for accessibility reasons.
+- Limit the image scale by picking a cut-off date.
+
+- Good: <https://www.php.net/supported-versions.php>
+- Good: <https://jefftriplett.com/django-release-cycle/>
+- Good: <https://hugovk.github.io/drupal-release-cycle/>
+- Bad: <https://docs.nvidia.com/datacenter/tesla/drivers/graphics/driver-branches-overview.png> (Cryptic)
+- Bad: <https://ubuntu.com/about/release-cycle> (Does not provide an accessible table)
+
+## Checklist
+
+Here's a nice checklist of all our recommendations. These are our recommendations - feel free to ignore what doesn't work for you.
+
+- [ ] Document all relevant information *together* in the same place.
+- [ ] Publish on a stable URL, and make sure this link is not versioned.
+- [ ] Document your release cadence
+- [ ] Explain all levels of support
+- [ ] Document your versioning policy
+- [ ] Demarcate your list of releases between supported and unsupported releases.
+- [ ] Provide the latest version in every support cycle
+- [ ] Provide absolute dates, instead of relative ones.
+- [ ] Provide complete dates, include the day of the month.
+- [ ] If you have a release schedule image, label it clearly and mark a current date line.
```

Comment:
```suggestion
- [ ] Document all relevant information *together* in the same place.
- [ ] Publish at a stable URL, and make sure this link is not versioned.
- [ ] Document your release cadence.
- [ ] Explain all levels of support.
- [ ] Document your versioning policy.
- [ ] Demarcate your list of releases between supported and unsupported releases.
- [ ] Provide the latest version in every support cycle.
- [ ] Provide absolute dates, instead of relative ones.
- [ ] Provide complete dates, include the day of the month.
- [ ] If you have a release schedule image, label it clearly and mark a current date line.
```